<details><summary>Relevant output</summary>

```
[76/271] Compiling C object 'sway/345c74e@@sway@exe/desktop_output.c.o'.
FAILED: sway/345c74e@@sway@exe/desktop_output.c.o
/nix/store/ds1prvgw3i3ic8c7axyrw4lwm3d0gqab-gcc-wrapper-7.4.0/bin/cc -Isway/345c74e@@sway@exe -Isway -I../sway -Iinclude -I../include -I/nix/store/chq3xpz7r54c9lh3xf8y330zsj1w4aw3-cairo-1.16.0-dev/include/cairo -I/nix/store/z6wjrmaflgkjj3ckzqc7d0ap79rhbi1j-freetype-2.10.0-dev/include/freetype2 -I/nix/store/z6wjrmaflgkjj3ckzqc7d0ap79rhbi1j-freetype-2.10.0-dev/include -I/nix/store/g2pjsgklp091m55gvvj721p33yljzzrd-json-c-0.13.1-dev/include/json-c -I/nix/store/9sni6rbv28hdfw7h0qqyrx1chswq1srg-libevdev-1.6.0/include/libevdev-1.0/ -I/nix/store/42yvfxq74vfxd2643x4fndpq7ndiirh7-libinput-1.13.2-dev/include -I/nix/store/1s97ja307db2dv5l1qvy4xzyw9nfmkb7-pango-1.43.0-dev/include/pango-1.0 -I/nix/store/0y6pfd81k1kxbrz1nnqjnir7vihl29ry-glib-2.60.2-dev/include -I/nix/store/0y6pfd81k1kxbrz1nnqjnir7vihl29ry-glib-2.60.2-dev/include/glib-2.0 -I/nix/store/aj9v5zj7v6i6rwhw6s1c60146ljspyb5-glib-2.60.2/lib/glib-2.0/include -I/nix/store/vhbm21ns6drmg9xgcpaqlfw88gxhw2rg-pcre-8.43-dev/include -I/nix/store/b0cahknmmfrlr1pd3mnmldy98v4xyh6l-pixman-0.38.4/include/pixman-1 -I/nix/store/r0qby1xczlyiglkph4gdwdj0sgp6i9n7-wayland-1.17.0/include -I/nix/store/z82mmy9l5zwmmvfj9ali6iplbw007lgy-wlroots-0.6.0/include -I/nix/store/rcmsp0p5m6cqdsivq5hx7j1dfgaqpmpv-libxkbcommon-0.8.0-dev/include -I/nix/store/h8aqv4ckyj99bbikd5rzn37n27sxf52i-libxcb-1.13.1-dev/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O3 '-DSWAY_SRC_DIR="/build/source"' -DWL_HIDE_DEPRECATED -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wundef -Wvla '-DSYSCONFDIR="/etc"' '-DNIX_SYSCONFDIR="//nix/store/13qdx1yixy4xak3klv60pkll90i2zz41-sway-1.1/etc"' '-DSWAY_VERSION="1.0"'  -MD -MQ 'sway/345c74e@@sway@exe/desktop_output.c.o' -MF 'sway/345c74e@@sway@exe/desktop_output.c.o.d' -o 'sway/345c74e@@sway@exe/desktop_output.c.o' -c ../sway/desktop/output.c
../sway/desktop/output.c: In function 'update_output_manager_config':
../sway/desktop/output.c:512:3: error: implicit declaration of function 'wlr_output_configuration_v1_create'; did you mean 'wlr_output_cursor_create'? [-Werror=implicit-function-declaration]
   wlr_output_configuration_v1_create();
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   wlr_output_cursor_create
../sway/desktop/output.c:512:3: error: initialization makes pointer from integer without a cast [-Werror=int-conversion]
../sway/desktop/output.c:520:4: error: implicit declaration of function 'wlr_output_configuration_head_v1_create'; did you mean 'wlr_output_cursor_create'? [-Werror=implicit-function-declaration]
    wlr_output_configuration_head_v1_create(config, output->wlr_output);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    wlr_output_cursor_create
../sway/desktop/output.c:520:4: error: initialization makes pointer from integer without a cast [-Werror=int-conversion]
../sway/desktop/output.c:524:14: error: dereferencing pointer to incomplete type 'struct wlr_output_configuration_head_v1'
   config_head->state.enabled = output->enabled;
              ^~
../sway/desktop/output.c:531:2: error: implicit declaration of function 'wlr_output_manager_v1_set_configuration'; did you mean 'wlr_output_transformed_resolution'? [-Werror=implicit-function-declaration]
  wlr_output_manager_v1_set_configuration(server->output_manager_v1, config);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  wlr_output_transformed_resolution
../sway/desktop/output.c:531:50: error: 'struct sway_server' has no member named 'output_manager_v1'; did you mean 'idle_inhibit_manager_v1'?
  wlr_output_manager_v1_set_configuration(server->output_manager_v1, config);
                                                  ^~~~~~~~~~~~~~~~~
                                                  idle_inhibit_manager_v1
cc1: all warnings being treated as errors
```

</details>

--- a/pkgs/applications/window-managers/sway/default.nix
+++ b/pkgs/applications/window-managers/sway/default.nix
@@ -1,35 +1,28 @@
-{ stdenv, fetchFromGitHub, fetchpatch
+{ stdenv, fetchFromGitHub, makeWrapper
 , meson, ninja
 , pkgconfig, scdoc
 , wayland, libxkbcommon, pcre, json_c, dbus, libevdev
 , pango, cairo, libinput, libcap, pam, gdk_pixbuf
-, wlroots, wayland-protocols
+, wlroots, wayland-protocols, swaybg
 }:
 
 stdenv.mkDerivation rec {
-  name = "${pname}-${version}";
   pname = "sway";
-  version = "1.0";
+  version = "1.1.1";
 
   src = fetchFromGitHub {
     owner = "swaywm";
     repo = "sway";
     rev = version;
-    sha256 = "09cndc2nl39d3l7g5634xp0pxcz60pvc5277mfw89r22mh0j78rx";
+    sha256 = "0yhn9zdg9mzfhn97c440lk3pw6122nrhx0is5sqmvgr6p814f776";
   };
 
   patches = [
-    # Fix for a compiler warning that causes a build failure
-    # (see https://github.com/swaywm/sway/issues/3862):
-    (fetchpatch {
-      url = "https://github.com/swaywm/sway/commit/bcde298a719f60b9913133dbd2a169dedbc8dd7d.patch";
-      sha256 = "0r583nmqvq43ib93yv6flw8pj833v32lbs0q0xld56s3rnzvvdcp";
-    })
     ./sway-config-no-nix-store-references.patch
     ./load-configuration-from-etc.patch
   ];
 
-  nativeBuildInputs = [ pkgconfig meson ninja scdoc ];
+  nativeBuildInputs = [ pkgconfig meson ninja scdoc makeWrapper ];
 
   buildInputs = [
     wayland libxkbcommon pcre json_c dbus libevdev
@@ -43,6 +36,10 @@ stdenv.mkDerivation rec {
     "-Ddefault-wallpaper=false" "-Dxwayland=enabled" "-Dgdk-pixbuf=enabled"
     "-Dtray=enabled" "-Dman-pages=enabled"
   ];
+
+  postInstall = ''
+    wrapProgram $out/bin/sway --prefix PATH : "${swaybg}/bin"
+  '';
 
   meta = with stdenv.lib; {
     description = "i3-compatible tiling Wayland compositor";

--- a/pkgs/applications/window-managers/sway/default.nix
+++ b/pkgs/applications/window-managers/sway/default.nix
@@ -41,6 +41,10 @@ stdenv.mkDerivation rec {
     wrapProgram $out/bin/sway --prefix PATH : "${swaybg}/bin"
   '';
 
+  postPatch = ''
+    sed -i "s/version: '1.0'/version: '${version}'/" meson.build
+  '';
+
   meta = with stdenv.lib; {
     description = "i3-compatible tiling Wayland compositor";
     homepage    = https://swaywm.org;

--- a/pkgs/development/libraries/wlroots/default.nix
+++ b/pkgs/development/libraries/wlroots/default.nix
@@ -4,17 +4,15 @@
 , libpng, ffmpeg_4
 }:
 
-let
+stdenv.mkDerivation rec {
   pname = "wlroots";
-  version = "0.5.0";
-in stdenv.mkDerivation rec {
-  name = "${pname}-${version}";
+  version = "0.6.0";
 
   src = fetchFromGitHub {
     owner = "swaywm";
     repo = "wlroots";
     rev = version;
-    sha256 = "1phiidyddzgaxy4gbqwmykxn0y8za6y5mp66l9dpd9i6fml153yq";
+    sha256 = "1rdcmll5b8w242n6yfjpsaprq280ck2jmbz46dxndhignxgda7k4";
   };
 
   # $out for the library, $bin for rootston, and $examples for the example

